lua/obsidian/command.lua

```diff
@@ -287,70 +295,73 @@ command.insert_template = function(client, data)
     vim.api.nvim_win_set_cursor(0, { new_row, 0 })
   end
 
-  -- try with telescope.nvim
-  local has_telescope, _ = pcall(require, "telescope.builtin")
-  if has_telescope then
-    local choose_template = function()
-      local opts = {
-        cwd = tostring(templates_dir),
-        attach_mappings = function(_, map)
-          map({ "i", "n" }, "<CR>", function(prompt_bufnr)
-            local template = require("telescope.actions.state").get_selected_entry()
-            require("telescope.actions").close(prompt_bufnr)
-            apply_template(template[1])
-          end)
-          return true
-        end,
-      }
-      require("telescope.builtin").find_files(opts)
-    end
-    choose_template()
-    return
-  end
-
-  -- try with fzf-lua
-  local has_fzf_lua, fzf_lua = pcall(require, "fzf-lua")
-  if has_fzf_lua then
-    local cmd = vim.tbl_flatten { util.FIND_CMD, { ".", "-name", "'*.md'" } }
-    cmd = util.table_params_to_str(cmd)
-    fzf_lua.files {
-      cmd = cmd,
-      cwd = tostring(templates_dir),
-      file_icons = false,
-      actions = {
-        ["default"] = function(entry)
-          -- for some reason fzf-lua passes the filename with 6 characters
-          -- at the start that appear on screen as 2 whitespace characters
-          -- so we need to start on the 7th character
-          local template = entry[1]:sub(7)
+  client:_run_with_finder_backend(":ObsidianTemplate", {
+    ["telescope.nvim"] = function()
+      -- try with telescope.nvim
+      local has_telescope, _ = pcall(require, "telescope.builtin")
+      if has_telescope then
+        local choose_template = function()
+          local opts = {
+            cwd = tostring(templates_dir),
+            attach_mappings = function(_, map)
+              map({ "i", "n" }, "<CR>", function(prompt_bufnr)
+                local template = require("telescope.actions.state").get_selected_entry()
+                require("telescope.actions").close(prompt_bufnr)
+                apply_template(template[1])
+              end)
+              return true
+            end,
+          }
+          require("telescope.builtin").find_files(opts)
+        end
+        choose_template()
+        return true
+      end
```

Comment:
I think we should explicitly return false if `has_telescope` is false, and same for the others below. Just for clarity.

Reply:
Should I also verify in `run_first_supported` that the returned value is boolean?

Reply:
Another option is to throw a magic value from skipped implementations. That way `run_first_supported` can return the value of the implementation that succeeded.

Reply:
> Another option is to throw a magic value from skipped implementations. That way run_first_supported can return the value of the implementation that succeeded.

I like that idea 👍